data_quality/generators/ckan.py

```diff
@@ -0,0 +1,123 @@
+# -*- coding: utf-8 -*-
+from __future__ import division
+from __future__ import print_function
+from __future__ import absolute_import
+from __future__ import unicode_literals
+
+import csv
+import requests
+from os import path
+from data_quality import compat
+from .base import BaseGenerator
+
+class CkanGenerator(BaseGenerator):
+    """This class generates a csv database from a CKAN instance located at the given url"""
+
+    def __init__(self, url=None):
+        """Create an instance if the source url is given.
+
+        Args:
+            url: the base url for the CKAN instance
+        """
+
+        super(CkanGenerator, self).__init__(url)
+
+    def generate_sources(self, sources_filepath, file_types=['csv','excel']):
+        """Generates sources_file from the url"""
+
+        fieldnames = ['id', 'publisher_id', 'title', 'data', 'format', 'period_id']
```

Comment:
We are maintaining lists of field names for all files in the database (including, as here, sources) in multiple places. For ease of refactoring in future, I suggest centralising these. They can be represented as [JSON Table Schemas](http://dataprotocols.org/json-table-schema/)